### PR TITLE
Tech/remove files class

### PR DESCRIPTION
**Pull request by NearW**
# Remove class Files and move the logic to helper functions

## Description

Regarding my [post ](https://github.com/MaibornWolff/codecharta/pull/943#issuecomment-621336574), I did the following things to adress this issue

- Moved all methods from `Files.ts` to helper functions in `files.helper.ts`
- Removed `Files.ts` and updated all references including changing all types from `Files` back to `FileState[]`
- Moved the `files.helper.ts` and parts of the `codecharta.model.ts` regarding fileStates to the model folder
- Removed TECHNICAL_DEBT.md since it's not in use at all and it only includes my `Blacklist as class` entry which is not possible in redux

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [x] **All** requirements mentioned in the issue are implemented
- [x] Does match the Code of Conduct and the Contribution file
- [ ] Task has its own **GitHub issue** (something it is solving)
  - [ ] Issue number is **included in the commit messages** for traceability
- [x] **Update the README.md** with any changes/additions made
- [x] **Update the CHANGELOG.md** with any changes/additions made
- [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [x] **All tests pass**
- [x] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [x] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts [hunk 1]
```diff
@@ -25,13 +26,13 @@ export class BlacklistService implements StoreSubscriber, FilesSelectionSubscrib
...
-	public onFilesSelectionChanged(files: Files) {
+	public onFilesSelectionChanged(files: FileState[]) {
 		this.merge(files)
 	}
 
-	private merge(files: Files) {
-		const visibleFiles = files.getVisibleFileStates().map(x => x.file)
-		const withUpdatedPath = files.isPartialState()
+	private merge(files: FileState[]) {
+		const visibleFiles = getVisibleFileStates(files).map(x => x.file)
```
Comment (alschmut) [inline visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts:34]:
```suggestion
		const visibleFiles = getVisibleFiles(files)
```

File: visualization/app/codeCharta/state/store/fileSettings/edges/edges.service.ts [hunk 2]
```diff
@@ -25,13 +26,13 @@ export class EdgesService implements StoreSubscriber, FilesSelectionSubscriber {
...
-	public onFilesSelectionChanged(files: Files) {
+	public onFilesSelectionChanged(files: FileState[]) {
 		this.merge(files)
 	}
 
-	private merge(files: Files) {
-		const visibleFiles = files.getVisibleFileStates().map(x => x.file)
-		const withUpdatedPath = files.isPartialState()
+	private merge(files: FileState[]) {
+		const visibleFiles = getVisibleFileStates(files).map(x => x.file)
```
Comment (alschmut) [inline visualization/app/codeCharta/state/store/fileSettings/edges/edges.service.ts:34]:
```suggestion
		const visibleFiles = getVisibleFiles(files)
```

**Comment by alschmut** [comment]
I was reviewing the changes and tried to check a lot in detail. I hope I didn't oversee something. Also the application works as expected IMO.

File: visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts (hunk 1 above)
Comment (NearW) [reply visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.service.ts:34]: nice catch, done

**Comment by NearW** [comment]
@alschmut Thanks for your review. The changes are pretty straight forward. The only thing that was a bit tricky were the changes in the .html files because there was no real IDE support.

**Pull request merged**